src/db/sqlite.cpp

```diff
@@ -319,7 +395,17 @@ void SqliteDbTable::commit()
     }
 
     LOG_debug << "DB transaction COMMIT " << dbfile;
-    sqlite3_exec(db, "COMMIT", 0, 0, NULL);
+
+    if (sqlite3_get_autocommit(db))
```

Comment:
Wouldn't it be a programmer's fault if the code calls `commit()` on a non-transactional mode DB?
This addition will hide that mistake.
I guess that you added the call because it's already occurring?

Reply:
Aye, I added the guard as commit() is being called unconditionally in some places.

Perhaps a better solution would have been to expose a function that allows the client to query whether the database is currently within a transaction. That way, we could call commit only if necessary.

That'd solve this problem nicely.